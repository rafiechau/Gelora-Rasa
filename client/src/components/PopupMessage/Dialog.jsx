import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { Dialog } from '@mui/material';

import classes from './style.module.scss';

// eslint-disable-next-line arrow-body-style
const PopupMessage = ({ open, title, message, onClose }) => {
  return (
    <Dialog open={open} onClose={onClose} PaperProps={{ className: classes.dialogWrapper }} data-testid="popup-message">
      <div className={classes.title}>
        <FormattedMessage id={title || 'app_popup_error_title'} />
      </div>
      <div className={classes.message}>
        <FormattedMessage id={message || 'app_popup_error_message'} />
      </div>
      <button type="button" onClick={onClose} className={classes.button} data-testid="popup-close-button">
        <FormattedMessage id="app_popup_close_button_label" />
      </button>
    </Dialog>
  );
};

PopupMessage.propTypes = {
  open: PropTypes.bool,
  title: PropTypes.string,
  message: PropTypes.string,
  onClose: PropTypes.func,
};

export default PopupMessage;
