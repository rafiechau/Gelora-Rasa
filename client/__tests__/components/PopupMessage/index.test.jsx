import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import PopupMessage from '@components/PopupMessage/Dialog';

// Contoh pesan dan judul untuk PopupMessage
const messages = {
  test_title: 'Test Title',
  test_message: 'Test Message',
  app_popup_close_button_label: 'Close',
};

const mockOnClose = jest.fn();

describe('PopupMessage Component', () => {
  beforeEach(() => {
    render(
      <IntlProvider locale="en" messages={messages}>
        <PopupMessage
          open
          onClose={mockOnClose}
          title="test_title"
          message="test_message"
          data-testid="popup-message"
        />
      </IntlProvider>
    );
  });

  test('should render PopupMessage', () => {
    expect(screen.getByTestId('popup-message')).toBeInTheDocument();
  });

  test('should call onClose when close button is clicked', () => {
    fireEvent.click(screen.getByTestId('popup-close-button'));
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    const popupMessage = screen.getByTestId('popup-message');
    expect(popupMessage).toMatchSnapshot();
  });
});
