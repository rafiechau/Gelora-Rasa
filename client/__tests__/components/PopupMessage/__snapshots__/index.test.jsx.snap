// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupMessage Component should match snapshot 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="popup-message"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r2:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm dialogWrapper css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <div
        class="title"
      >
        Test Title
      </div>
      <div
        class="message"
      >
        Test Message
      </div>
      <button
        class="button"
        data-testid="popup-close-button"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
